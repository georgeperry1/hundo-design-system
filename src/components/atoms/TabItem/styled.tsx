import styled from 'styled-components';

type DefaultTabType = {
  active?: boolean;
  hideBorder?: boolean;
  disabled?: boolean;
};

export const DefaultTab = styled.div<DefaultTabType>`
  display: flex;
  width: max-content;
  align-items: center;
  justify-content: center;
  min-height: 35px;
  padding: 0 12px;
  cursor: ${({ disabled }) => (disabled ? 'not-allowed' : 'pointer')};
  opacity: ${({ disabled }) => (disabled ? '40%' : '100%')};
  border-bottom: 4px solid;
  border-bottom-color: ${({ theme, active, hideBorder }) => {
    if (hideBorder) return theme.colors.fullWhite;

    return active ? theme.colors.fullBlack : theme.colors.fullWhite;
  }};

  &:hover {
    border-bottom-color: ${({ theme }) => theme.colors.fullBlack};
  }
`;

export const TabText = styled.p`
  font-family: ${({ theme }) => theme.fonts.secondary};
  font-size: ${({ theme }) => theme.fontSizes[1]};
  font-weight: 500;
  line-height: 24px;
  color: ${({ theme }) => theme.colors.fullBlack};
`;
