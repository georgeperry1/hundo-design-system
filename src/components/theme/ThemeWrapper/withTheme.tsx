import React from 'react';
import { ThemeProvider } from 'styled-components';
import { lightTheme } from '../themes';

export function withTheme<P>(WrappedComponent: React.ComponentType<P>): React.FC<P> {
  const ComponentWithTheme = (props: P) => {
    return (
      <ThemeProvider theme={lightTheme}>
        <WrappedComponent {...props} />
      </ThemeProvider>
    );
  };
  return ComponentWithTheme;
}
export default withTheme;
