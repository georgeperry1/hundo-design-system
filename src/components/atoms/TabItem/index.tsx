import React from 'react';

import { Tab, TabText } from './styled';
import { TabVariant } from '../../molecules/Tabs/types';

export type TabItemProps = {
  value: string;
  label: string;
  disabled?: boolean;
  active?: boolean;
  icon?: string;
  onTabClick: () => void;
  hideBorder?: boolean;
  variant?: string;
  totalTabs: number;
};

const TabItem: React.FC<TabItemProps> = ({ onTabClick, label, disabled, variant, icon, ...props }) => {
  const onClick = () => {
    if (disabled) return;

    onTabClick();
  };

  return (
    <Tab onClick={onClick} variant={variant} {...props}>
      {icon && <img src={icon} alt={label} />}
      <TabText variant={variant}>{label}</TabText>
    </Tab>
  );
};

TabItem.defaultProps = {
  active: false,
  hideBorder: false,
  disabled: false,
  variant: TabVariant.Primary,
};

export default TabItem;
