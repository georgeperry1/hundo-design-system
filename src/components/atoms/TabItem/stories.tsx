import { Story, Meta } from '@storybook/react';

import TabItem, { TabItemProps } from '.';
import { TabVariant } from '../../molecules/Tabs/types';

export default {
  title: 'Components/Atoms/TabItem',
  component: TabItem,
} as Meta;

const Template: Story<TabItemProps> = (args) => <TabItem {...args} />;

export const Default = Template.bind({});

Default.args = {
  value: 'Hundo',
  label: 'Hundo',
  active: false,
  hideBorder: false,
};

export const Secondary: Story<TabItemProps> = (args) => (
  <div style={{ background: 'black' }}>
    <TabItem {...args} />
  </div>
);
Secondary.args = {
  value: 'Hundo',
  label: 'Hundo',
  active: false,
  hideBorder: false,
  variant: TabVariant.Secondary,
};

export const Active = Template.bind({});

Active.args = {
  value: 'Hundo',
  label: 'Hundo',
  active: true,
  hideBorder: false,
};

export const Disabled = Template.bind({});

Disabled.args = {
  value: 'Hundo',
  label: 'Hundo',
  active: true,
  hideBorder: false,
  disabled: true,
};
