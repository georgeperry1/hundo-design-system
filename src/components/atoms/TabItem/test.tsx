import { render, fireEvent, screen } from '@testing-library/react';

import TabItem from '.';
import ThemeWrapper from '../../../utils/test-theme-wrapper';

describe('Spacer', () => {
  it('should render', () => {
    const { baseElement } = render(
      <ThemeWrapper TestedComponent={TabItem} componentProps={{ value: 'Tab 1', label: 'Tab 1' }} />,
    );

    expect(baseElement).toMatchSnapshot();
  });

  it('should show a bottom border when active', () => {
    const { baseElement } = render(
      <ThemeWrapper TestedComponent={TabItem} componentProps={{ value: 'Tab 1', label: 'Tab 1', active: true }} />,
    );

    expect(baseElement).toMatchSnapshot();
  });

  it('should hide the border', () => {
    const { baseElement } = render(
      <ThemeWrapper
        TestedComponent={TabItem}
        componentProps={{ value: 'Tab 1', label: 'Tab 1', active: true, hideBorder: true }}
      />,
    );

    expect(baseElement).toMatchSnapshot();
  });

  it('should send event when clicked', () => {
    const handleClick = jest.fn();

    render(
      <ThemeWrapper
        TestedComponent={TabItem}
        componentProps={{ value: 'Tab 1', label: 'Tab 1', onTabClick: handleClick }}
      />,
    );

    fireEvent.click(screen.getByText('Tab 1'));

    expect(handleClick).toHaveBeenCalled();
  });

  it('should not send click event if disabled', () => {
    const handleClick = jest.fn();

    render(
      <ThemeWrapper
        TestedComponent={TabItem}
        componentProps={{ value: 'Tab 2', label: 'Tab 2', onTabClick: handleClick, disabled: true }}
      />,
    );

    fireEvent.click(screen.getByText('Tab 2'));

    expect(handleClick).not.toHaveBeenCalled();
  });
});
