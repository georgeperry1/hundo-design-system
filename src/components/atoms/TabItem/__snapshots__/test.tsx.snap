// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spacer should hide the border 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  min-height: 35px;
  padding: 0 12px;
  cursor: pointer;
  opacity: 100%;
  border-bottom: 4px solid;
  border-bottom-color: #ffffff;
}

.c0:hover {
  border-bottom-color: #000000;
}

.c1 {
  font-family: Inter;
  font-size: 14;
  font-weight: 500;
  line-height: 24px;
  color: #000000;
}

<body>
  <div>
    <div
      class="c0"
      value="Tab 1"
    >
      <p
        class="c1"
      >
        Tab 1
      </p>
    </div>
  </div>
</body>
`;

exports[`Spacer should render 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  min-height: 35px;
  padding: 0 12px;
  cursor: pointer;
  opacity: 100%;
  border-bottom: 4px solid;
  border-bottom-color: #ffffff;
}

.c0:hover {
  border-bottom-color: #000000;
}

.c1 {
  font-family: Inter;
  font-size: 14;
  font-weight: 500;
  line-height: 24px;
  color: #000000;
}

<body>
  <div>
    <div
      class="c0"
      value="Tab 1"
    >
      <p
        class="c1"
      >
        Tab 1
      </p>
    </div>
  </div>
</body>
`;

exports[`Spacer should show a bottom border when active 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  min-height: 35px;
  padding: 0 12px;
  cursor: pointer;
  opacity: 100%;
  border-bottom: 4px solid;
  border-bottom-color: #000000;
}

.c0:hover {
  border-bottom-color: #000000;
}

.c1 {
  font-family: Inter;
  font-size: 14;
  font-weight: 500;
  line-height: 24px;
  color: #000000;
}

<body>
  <div>
    <div
      class="c0"
      value="Tab 1"
    >
      <p
        class="c1"
      >
        Tab 1
      </p>
    </div>
  </div>
</body>
`;
